src_c/polygon.c

```diff
@@ -305,6 +305,131 @@ pg_polygon_get_vertices(pgPolygonObject *self, void *closure)
     return _pg_polygon_vertices_aslist(&self->polygon);
 }
 
+static int
+pg_polygon_ass_vertex(pgPolygonObject *self, Py_ssize_t i, PyObject *v)
+{
+    pgPolygonBase *poly = &self->polygon;
+
+    if (i < 0) {
+        i += poly->verts_num;
+    }
+
+    if (i < 0 || i > poly->verts_num) {
+        PyErr_SetString(PyExc_IndexError, "Invalid vertex Index");
+        return -1;
+    }
+
+    double v1, v2;
+    if (!pg_TwoDoublesFromObj(v, &v1, &v2)) {
+        PyErr_SetString(PyExc_TypeError, "Must assign numeric values");
+        return -1;
+    }
+    poly->vertices[i * 2] = v1;
+    poly->vertices[i * 2 + 1] = v2;
+    return 0;
+}
+
+static Py_ssize_t
+pg_polygon_seq_length(pgPolygonObject *self)
+{
+    pgPolygonBase *poly = &self->polygon;
+    return poly->verts_num;
+}
+
+static int
+pg_polygon_ass_subscript(pgPolygonObject *self, PyObject *op, PyObject *value)
+{
+    if (PyIndex_Check(op)) {
+        PyObject *index;
+        Py_ssize_t i;
+
+        index = PyNumber_Index(op);
+        if (index == NULL) {
+            return -1;
+        }
+        i = PyNumber_AsSsize_t(index, NULL);
+        Py_DECREF(index);
+        return pg_polygon_ass_vertex(self, i, value);
+    }
+    else {
+        PyErr_SetString(PyExc_TypeError, "Expected a number or sequence");
+        return -1;
+    }
+}
+
+static PyObject *
+pg_polygon_subscript(pgPolygonObject *self, PyObject *op)
+{
+    PyObject *vertices = _pg_polygon_vertices_aslist(&self->polygon);
+    PyObject *index = PyNumber_Index(op);
+    Py_ssize_t i;
+
+    if (index == NULL) {
+        return NULL;
+    }
+    i = PyNumber_AsSsize_t(index, NULL);
+    Py_DECREF(index);
+
+    pgPolygonBase *poly = &self->polygon;
+
+    if (i < 0) {
+        i += poly->verts_num;
+    }
+
+    if (i < 0 || i > poly->verts_num) {
+        return RAISE(PyExc_IndexError, "Invalid vertex Index");
+    }
+
+    return PyList_GetItem(vertices, i);
+}
+
+static int
+pg_polygon_contains_seq(pgPolygonObject *self, PyObject *arg)
+{
+    pgPolygonBase *poly = &self->polygon;
+
+    if (PyTuple_Check(arg) || PyList_Check(arg)) {
```

Comment:
Use `PySequence_FAST_CHECK` here
```suggestion
    if (PySequence_FAST_CHECK (arg)) {
```

Reply:
good catch